File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let toRna (dna: string): string =
    let complement b =
        match b with
        | 'G' -> 'C'
        | 'C' -> 'G'
        | 'T' -> 'A'
        | 'A' -> 'U'
        | _ -> raise (System.ArgumentException("Invalid base"))
    
    String.map complement dna
    

File: ocaml/rna-transcription/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna dna =
    let complement b =
        match b with
        | `A -> `U
        | `C -> `G
        | `G -> `C
        | `T -> `A
    in
    List.map complement dna
